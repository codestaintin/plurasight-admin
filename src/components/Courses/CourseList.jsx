import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CourseListRow from './CourseListRow';

const CourseList = ({ courses, isLoading, handleDelete }) => (
  <Fragment>
    <table className="table">
      <thead>
        <tr>
          <th>&nbsp;</th>
          <th>Title</th>
          <th>Author</th>
          <th>Category</th>
          <th>Length</th>
          <th>Actions</th>
        </tr>
      </thead>
      <tbody>
        {
          courses.map(course => (
            <CourseListRow
              key={course.id}
              course={course}
              handleDelete={handleDelete}
            />
          )
          )}
      </tbody>
    </table>
  </Fragment>

);

CourseList.propTypes = {
  courses: PropTypes.array.isRequired,
  isLoading: PropTypes.bool,
  handleDelete: PropTypes.func.isRequired
};

export default CourseList;
