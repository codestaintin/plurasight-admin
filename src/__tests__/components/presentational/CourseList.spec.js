import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import CourseList from '../../../components/Courses/CourseList';

configure({ adapter: new Adapter() });

describe('<CourseList>', () => {
  const props = {
    courses: [],
    handleDelete: jest.fn(() => {}),
  };

  it('renders the <CourseList> component', () => {
    const tree = shallow(<CourseList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
