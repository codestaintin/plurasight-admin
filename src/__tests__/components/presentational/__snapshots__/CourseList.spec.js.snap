// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseList> renders the <CourseList> component 1`] = `
<Fragment>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Â 
        </th>
        <th>
          Title
        </th>
        <th>
          Author
        </th>
        <th>
          Category
        </th>
        <th>
          Length
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
